src/aot/aot_main.cpp

```diff
@@ -95,7 +95,14 @@ int main(int argc, char* argv[])
     }
   }
 
-  if (!argv[optind]) {
+  if (argv[optind]) {
+    usage();
+    return 1;
+  }
+
+  if (strcmp(argv[0], aot::AOT_SHIM_NAME) == 0) {
+    LOG(INFO) << "Runtime shim should not be run directly, please generate a "
+                 "binary using --aot option in bpftrace";
     usage();
```

Comment:
Printing `usage()` here seems confusing. This particular error is not an issue with cli flags.